src/pages-and-resources/app-settings-modal/AppSettingsModal.jsx

```diff
@@ -0,0 +1,111 @@
+import AppConfigFormDivider from '../discussions/app-config-form/apps/shared/AppConfigFormDivider';
+
+function AppSettingsModal({
+  intl, settings, title, children, onClose, onApply, enableAppLabel,
+  enableAppHelp, learnMoreURL, learnMoreText,
+}) {
+  const dispatch = useDispatch();
+  const learnMoreLink = (
+    <Hyperlink destination={learnMoreURL || 'https://www.google.com'} target="_blank">{learnMoreText}</Hyperlink>
+  );
+  return (
+    <ModalLayer
+      isOpen
+      closeText={intl.formatMessage(messages.cancel)}
+      dialogClassName="modal-dialog-centered modal-lg"
+      onClose={onClose}
+    >
+      <div
+        role="dialog"
+        aria-label={title}
+        className={classNames(
+          'bg-white',
+          'd-flex',
+          'flex-column',
+          'mw-xs',
+          'p-3',
+        )}
+      >
+        <h3>{title}</h3>
+        <AppConfigFormDivider />
+        <Form onSubmit={onApply}>
+          <FormSwitchGroup
+            id={`enable-${settings.id}-toggle`}
+            onChange={(e) => {
+              dispatch(enableDisableApp('', settings.id, e.target.value));
+            }}
+            onBlur={null}
+            defaultChecked={settings.isEnabled}
+            label={(
+              [
+                `${enableAppLabel} `,
+                <EnabledDisabledBadge enabled={settings.isEnabled} />,
+              ]
+            )}
+            helpText={
+              (
+                <p>
+                  {enableAppHelp}
+                  <br />
+                  {learnMoreLink}
+                </p>
+              )
+            }
+          />
+          <AppConfigFormDivider />
+          <TransitionReplace>
+            {settings.isEnabled
+              ? <React.Fragment key="app-enabled">{children}</React.Fragment>
+              : <React.Fragment key="app-disabled" />}
+          </TransitionReplace>
+        </Form>
+        {settings.isEnabled && children && <AppConfigFormDivider />}
+        <div className="d-flex justify-content-end">
+          <Button variant="cancel" className="btn btn-link" onClick={onClose}>
+            {intl.formatMessage(messages.cancel)}
+          </Button>
+          <Button
+            variant="success"
+            data-autofocus
+            onClick={() => {
+              if (onApply) { onApply(); }
+              onClose();
+            }}
+          >
+            {intl.formatMessage(messages.apply)}
+          </Button>
+        </div>
+      </div>
+    </ModalLayer>
+  );
+}
+
+AppSettingsModal.propTypes = {
+  intl: intlShape.isRequired,
+  title: PropTypes.string.isRequired,
+  settings: CoursePageShape.isRequired,
+  children: PropTypes.element.isRequired,
+  onClose: PropTypes.func.isRequired,
+  onApply: PropTypes.func.isRequired,
+  enableAppLabel: PropTypes.string.isRequired,
+  enableAppHelp: PropTypes.string.isRequired,
+  learnMoreURL: PropTypes.string.isRequired,
+  learnMoreText: PropTypes.string.isRequired,
```

Comment:
can we decrease the number of props passed by creating an object for all messages, links props?

Reply:
We should be using them, but I just don't have the data at the moment. The URL for where these docs will be hosted will be available once the docs exist. 

As for reducing the number of props. We can do that, but it won't help much since we'll still need to construct that object on the input side. This isn't an existing structure we can just pass on.